src/unix.rs

```diff
@@ -64,39 +64,52 @@ impl Client {
             limit -= n;
         }
 
-        set_nonblocking(write.as_raw_fd(), false)?;
+        set_nonblocking(write.as_fd(), false)?;
 
         Ok(client)
     }
 
-    unsafe fn mk() -> io::Result<Client> {
-        let mut pipes = [0; 2];
-
+    fn mk() -> io::Result<Client> {
         // Attempt atomically-create-with-cloexec if we can on Linux,
         // detected by using the `syscall` function in `libc` to try to work
         // with as many kernels/glibc implementations as possible.
         #[cfg(target_os = "linux")]
         {
             static PIPE2_AVAILABLE: AtomicBool = AtomicBool::new(true);
             if PIPE2_AVAILABLE.load(Ordering::SeqCst) {
-                match libc::syscall(libc::SYS_pipe2, pipes.as_mut_ptr(), libc::O_CLOEXEC) {
-                    -1 => {
-                        let err = io::Error::last_os_error();
-                        if err.raw_os_error() == Some(libc::ENOSYS) {
-                            PIPE2_AVAILABLE.store(false, Ordering::SeqCst);
-                        } else {
-                            return Err(err);
+                unsafe {
+                    let mut pipes = [0; 2];
+                    match libc::syscall(libc::SYS_pipe2, pipes.as_mut_ptr(), libc::O_CLOEXEC) {
+                        -1 => {
+                            let err = io::Error::last_os_error();
+                            if err.raw_os_error() == Some(libc::ENOSYS) {
+                                PIPE2_AVAILABLE.store(false, Ordering::SeqCst);
+                            } else {
+                                return Err(err);
+                            }
+                        }
+                        _ => {
+                            return Ok(Client::from_fds(
+                                OwnedFd::from_raw_fd(pipes[0]),
+                                OwnedFd::from_raw_fd(pipes[1]),
+                            ))
                         }
                     }
-                    _ => return Ok(Client::from_fds(pipes[0], pipes[1])),
                 }
             }
         }
 
-        cvt(libc::pipe(pipes.as_mut_ptr()))?;
-        drop(set_cloexec(pipes[0], true));
-        drop(set_cloexec(pipes[1], true));
-        Ok(Client::from_fds(pipes[0], pipes[1]))
+        let (read, write) = unsafe {
+            let mut pipes = [0; 2];
+            cvt(libc::pipe(pipes.as_mut_ptr()))?;
+            (
+                OwnedFd::from_raw_fd(pipes[0]),
+                OwnedFd::from_raw_fd(pipes[1]),
+            )
+        };
+        drop(set_cloexec(read.as_fd(), true));
+        drop(set_cloexec(write.as_fd(), true));
```

Comment:
I think the error should be propagated instead of ignored:

```suggestion
        set_cloexec(read.as_fd(), true)?;
        set_cloexec(write.as_fd(), true)?;
```